### ISSUE TITLE
BarChartView doesn't works as expected 

### ISSUE THREAD

**Issue by droidwala** [diogobernardino/WilliamChart#122]
BarChartView doesn't works as expected when using 
`mChart.setAxisBorderValues(int MIN,int MAX,int STEP)`  with MIN value other than 0(especially greater than 0).

Graph looks like when MIN > 0 : 
![non_zero_min_value](https://cloud.githubusercontent.com/assets/2155572/14140226/6ad03242-f695-11e5-8115-6ff73fda2c2e.jpg)         

As can be seen above rect(bar) starts from below x-axis.
`mChart.setAxisBorderValues(100,200,100);
`

But,
When same graph is plotted with MIN = 0:
![min_value_0](https://cloud.githubusercontent.com/assets/2155572/14140461/b4ee44da-f696-11e5-94d0-03fd2ab39ed5.jpg)

It works as designed!
`mChart.setAxisBorderValues(0,200,100)`

Is there anything in `onDrawChart(Canvas canvas,ArrayList<Charset> data)`  method causing this behave to this way.
Most probably last parameter passed to the `drawBar(Canvas canvas,float left,float top,float right,float bottom)`  inside onDrawChart method looks like the cause of issue.

P.S : When using other charts like LineChart i don't face  the same issue when using `setAxisBorderValues()` with varying MIN values.

:+1:  for the awesome library!

**Comment by diogobernardino** [comment]
I think I never considered such a case, where you have, for example, values from 0 to 200 but then you set an axis from 100 to 200.

Need to review those cases. Thanks for reporting.

**Comment by droidwala** [comment]
Actually my use case is like below:
I get weekly data values from server(5 float values for Mon-Fri).
Now, this values could be anything like 
case 1:
values between 50- 60
case 2:
values between 700-800
case 3:
values between 2000-2100

Now, when i don't use `setAxisBorderValues()` method in such cases (graph for values between 700-800).   

Graph looks like below:
![img_20160330_174018](https://cloud.githubusercontent.com/assets/2155572/14141762/b0c1add6-f69e-11e5-8fd5-88db8803bcec.jpg) 
It all becomes messy!

So,for all such above cases I first find out Min and Max values in float[] values(array) I use in my dataset.    
Then find STEP by doing GCD(Min,Max).    
Then use those MIN,MAX&STEP values calculated to get the graph starting from MIN value instead of default 0 value.

So, TLDR; version of the issue is:
**when my value range is from A to B(excluding A & B) and when I use 'A' as MIN value(>0) it starts from below x-axis.**

Thanks for quick response!

**Comment by diogobernardino** [comment]
Strange, I recall implementing GCD in case there was axis border values previously defined. It has been a long time I do not look at the code. Will keep the issue open and test it once I have the time.
